### PR TITLE
python3Packages.numpy: proper fix for Werror misdetection bug

### PR DESCRIPTION
**Pull request by veprbl**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
Builds happily on macos 10.15, python 3.8 & 3.9

**Review by risicle — APPROVED** [review]

File: pkgs/development/python-modules/numpy/default.nix [hunk 1]
```diff
@@ -50,7 +51,13 @@ in buildPythonPackage rec {
     sha256 = "1za22ggjxzm7drs2vd15s81ad9rlshk4p7pv7mxcbz4acdiszx6z";
   };
 
-  patches = lib.optionals python.hasDistutilsCxxPatch [
+  patches = [
+    # fix https://github.com/numpy/numpy/issues/19624
+    (fetchpatch {
+      url = "https://github.com/numpy/numpy/commit/ae279066d6bd253e8675428fac8946938b8d48d9.diff";
+      sha256 = "sha256-q8SoF/pMzLI4zYCx2YAZzR31cARlZXerNnBSevBO5pE=";
```
Comment (samuela) [inline pkgs/development/python-modules/numpy/default.nix:58]: `sha256 =` could be replaced with `hash =` here.

**Review by samuela — APPROVED** [review]

File: pkgs/development/python-modules/numpy/default.nix (hunk 1 above)
Comment (veprbl) [reply pkgs/development/python-modules/numpy/default.nix:58]: That wouldn't work yet
https://github.com/NixOS/nixpkgs/blob/722c8c9529baadd73fb0e5cf1ec896ca2a90b8f3/pkgs/build-support/fetchpatch/default.nix#L16

Comment (samuela) [reply pkgs/development/python-modules/numpy/default.nix:58]: Oh we should add `hash` support to fetchpatch then

**Review by r-burns — APPROVED** [review]
LGTM

**Pull request merged**